app/assets/javascripts/discourse/controllers/topic.js.es6

```diff
@@ -649,11 +649,37 @@ export default Ember.Controller.extend(SelectedPostsCount, BufferedContent, {
       const quotedText = Quote.build(post, quoteState.get('buffer'));
       this.send('deselectText');
 
-      composerController.open({
-        action: Composer.CREATE_TOPIC,
-        draftKey: Composer.REPLY_AS_NEW_TOPIC_KEY,
-        categoryId: this.get('model.category.id')
-      }).then(() => {
+      var options;
+
+      if (this.get('model.isPrivateMessage')) {
+
+        var users = this.get('model.details.allowed_users');
+        var groups = this.get('model.details.allowed_groups');
+
+        var usernames = users.map(user => {
+          return user.username;
+        }).concat(groups.map(group => {
+          return group.name;
+        })).join();
```

Comment:
Instead of joining at the end, I think we can just push the usernames into a string as we iterate each array and thus avoiding creating a new array.